From: Michael Froman <mfroman@mozilla.com>
Date: Sat, 17 Jan 2026 14:01:06 -0600
Subject: (tmp-cherry-pick) Revert "Make SignalTrampoline immovable"
 (54170ce987)

This reverts commit d50faaa9c53d6be93ec8340adfd51c9ac754866f.

Reason for revert: Chromium wasn't ready for it.

Original change's description:
> Make SignalTrampoline immovable
>
> This safeguards further deployments of SignalTrampoline, so that
> errors are not introduced.
>
> Bug: webrtc:42222066
> Change-Id: I34c2170a48305729def2da6b59f8c6674601f1d8
> No-IWYU: bot broken
> Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/423900
> Reviewed-by: Tomas Gunnarsson <tommi@webrtc.org>
> Commit-Queue: Harald Alvestrand <hta@webrtc.org>
> Cr-Commit-Position: refs/heads/main@{#46157}

No-IWYU: Bot broken
Bug: webrtc:42222066
Change-Id: Ibf235b38509e732f0e84c8d6f3107f3e63ef5bd6
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/424940
Auto-Submit: Harald Alvestrand <hta@webrtc.org>
Commit-Queue: Harald Alvestrand <hta@webrtc.org>
Reviewed-by: Danil Chapovalov <danilchap@webrtc.org>
Bot-Commit: Rubber Stamper <rubber-stamper@appspot.gserviceaccount.com>
Commit-Queue: Rubber Stamper <rubber-stamper@appspot.gserviceaccount.com>
Reviewed-by: Tomas Gunnarsson <tommi@webrtc.org>
Cr-Commit-Position: refs/heads/main@{#46168}
---
 rtc_base/sigslot_trampoline.h | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/rtc_base/sigslot_trampoline.h b/rtc_base/sigslot_trampoline.h
index f99e14c602..ccd0fca882 100644
--- a/rtc_base/sigslot_trampoline.h
+++ b/rtc_base/sigslot_trampoline.h
@@ -134,12 +134,6 @@ class SignalTrampoline
   using Base = internal::SignalTrampolineMemberBase<T, member_signal>;
 
  public:
-  // Because a SignalTrampoline is initialized with a pointer to its
-  // container, it's not possible to move or copy it.
-  SignalTrampoline(const SignalTrampoline&) = delete;
-  SignalTrampoline& operator=(const SignalTrampoline&) = delete;
-  SignalTrampoline(SignalTrampoline&&) = delete;
-  SignalTrampoline& operator=(SignalTrampoline&&) = delete;
   explicit SignalTrampoline(T* that) {
     (that->*member_signal).connect(static_cast<Base*>(this), &Base::Notify);
   }
